### ISSUE TITLE
failed upload assets to s3 & image not show

### ISSUE THREAD

**Issue by denzfarid** [grokability/snipe-it#8732]
#### Please confirm you have done the following before posting your bug report:

- [ ] I have enabled debug mode 
- [*] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

**Describe the bug**
Hello, I had an error when I tried to move assets to s3 bucket, I used the php artisan snipeit command: move-uploads, according to this directive
`php artisan snipeit:move-uploads`
https://snipe-it.readme.io/docs/configuration#optional-amazon-s3-settings

**To Reproduce**
I have replaced the env file like this

```
PUBLIC_FILESYSTEM_DISK : s3_public
PRIVATE_FILESYSTEM_DISK : s3_private


# --------------------------------------------
# OPTIONAL: PUBLIC S3 Settings
# --------------------------------------------
PUBLIC_AWS_SECRET_ACCESS_KEY=xxxxx
PUBLIC_AWS_ACCESS_KEY_ID=xxxxx
PUBLIC_AWS_DEFAULT_REGION=xxxxx
PUBLIC_AWS_BUCKET=xxxxx
PUBLIC_AWS_URL=xxxxx
PUBLIC_AWS_BUCKET_ROOT=storage

# --------------------------------------------
# OPTIONAL: PRIVATE S3 Settings
# --------------------------------------------
PRIVATE_AWS_ACCESS_KEY_ID=xxxxx
PRIVATE_AWS_SECRET_ACCESS_KEY=xxxxx
PRIVATE_AWS_DEFAULT_REGION=xxxxx
PRIVATE_AWS_BUCKET=xxxxx
PRIVATE_AWS_URL=xxxxx
PRIVATE_AWS_BUCKET_ROOT=storage

```
**Expected behavior**
but when I run the command, I get an error when want to upload a logo,
even though I have uploaded a logo, are there any steps that I missed?

**Screenshots**
<img width="507" alt="image" src="https://user-images.githubusercontent.com/1294403/98999138-d88f5d00-2569-11eb-8fa0-5aa90e8764c4.png">
<img width="427" alt="image" src="https://user-images.githubusercontent.com/1294403/98999189-f066e100-2569-11eb-847d-40f7aebdd287.png">

and this is when I change the asset to s3, all the images seem to fail to load
<img width="202" alt="image" src="https://user-images.githubusercontent.com/1294403/99000369-bdbde800-256b-11eb-90b6-492bf119e0b7.png">
but when I right click and view the image, the image can be loaded like this
<img width="559" alt="image" src="https://user-images.githubusercontent.com/1294403/99000337-b0086280-256b-11eb-8804-a1ac369f7031.png">

**Server (please complete the following information):**
 - Snipe-IT Version latest 
 - OS: ubuntu
 - Web Server: apache
 - PHP Version : 7.4.x

**Error Messages**
```
[2020-11-13 04:15:07] production.ERROR: ErrorException: Undefined offset: 2 in /var/www/snipeit/app/Console/Commands/MoveUploadsToNewDisk.php:94
Stack trace:
#0 /var/www/snipeit/app/Console/Commands/MoveUploadsToNewDisk.php(94): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError()
#1 [internal function]: App\Console\Commands\MoveUploadsToNewDisk->handle()
#2 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(32): call_user_func_array()
#3 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/Util.php(36): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#4 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(90): Illuminate\Container\Util::unwrapIfClosure()
#5 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(34): Illuminate\Container\BoundMethod::callBoundMethod()
#6 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/Container.php(590): Illuminate\Container\BoundMethod::call()
#7 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Console/Command.php(134): Illuminate\Container\Container->call()
#8 /var/www/snipeit/vendor/symfony/console/Command/Command.php(255): Illuminate\Console\Command->execute()
#9 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Console/Command.php(121): Symfony\Component\Console\Command\Command->run()
#10 /var/www/snipeit/vendor/symfony/console/Application.php(1001): Illuminate\Console\Command->run()
#11 /var/www/snipeit/vendor/symfony/console/Application.php(271): Symfony\Component\Console\Application->doRunCommand()
#12 /var/www/snipeit/vendor/symfony/console/Application.php(147): Symfony\Component\Console\Application->doRun()
#13 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Console/Application.php(93): Symfony\Component\Console\Application->run()
#14 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(131): Illuminate\Console\Application->run()
#15 /var/www/snipeit/artisan(35): Illuminate\Foundation\Console\Kernel->handle()
#16 {main}
[2020-11-13 04:15:07] production.ERROR: Undefined offset: 2 {"exception":"[object] (ErrorException(code: 0): Undefined offset: 2 at /var/www/snipeit/app/Console/Commands/MoveUploadsToNewDisk.php:94)
[stacktrace]
#0 /var/www/snipeit/app/Console/Commands/MoveUploadsToNewDisk.php(94): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError()
#1 [internal function]: App\\Console\\Commands\\MoveUploadsToNewDisk->handle()
#2 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(32): call_user_func_array()
#3 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/Util.php(36): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#4 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(90): Illuminate\\Container\\Util::unwrapIfClosure()
#5 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(34): Illuminate\\Container\\BoundMethod::callBoundMethod()
#6 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/Container.php(590): Illuminate\\Container\\BoundMethod::call()
#7 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Console/Command.php(134): Illuminate\\Container\\Container->call()
#8 /var/www/snipeit/vendor/symfony/console/Command/Command.php(255): Illuminate\\Console\\Command->execute()
#9 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Console/Command.php(121): Symfony\\Component\\Console\\Command\\Command->run()
#10 /var/www/snipeit/vendor/symfony/console/Application.php(1001): Illuminate\\Console\\Command->run()
#11 /var/www/snipeit/vendor/symfony/console/Application.php(271): Symfony\\Component\\Console\\Application->doRunCommand()
#12 /var/www/snipeit/vendor/symfony/console/Application.php(147): Symfony\\Component\\Console\\Application->doRun()
#13 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Console/Application.php(93): Symfony\\Component\\Console\\Application->run()
#14 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(131): Illuminate\\Console\\Application->run()
#15 /var/www/snipeit/artisan(35): Illuminate\\Foundation\\Console\\Kernel->handle()
#16 {main}
"}
```
Thank you :)

**Comment by snipe** [comment]
S3 support is still highly experimental. We'll take a look at this.

**Comment by snipe** [comment]
I've pushed out a possible fix onto the develop branch

**Comment by denzfarid** [comment]
> S3 support is still highly experimental. We'll take a look at this.

Thank you in advance

> I've pushed out a possible fix onto the develop branch

perfect, 
I tried checkout to the develop branch and the logo problem has been resolved but there are still errors on the same line but the contents are different

<img width="375" alt="image" src="https://user-images.githubusercontent.com/1294403/99007101-1c885f00-2576-11eb-9570-53118dd4cd42.png">

```
[2020-11-13 05:52:53] production.ERROR: Undefined offset: 2 {"exception":"[object] (ErrorException(code: 0): Undefined offset: 2 at /var/www/snipeit/app/Console/Commands/MoveUploadsToNewDisk.php:94)
[stacktrace]
#0 /var/www/snipeit/app/Console/Commands/MoveUploadsToNewDisk.php(94): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError()
#1 [internal function]: App\\Console\\Commands\\MoveUploadsToNewDisk->handle()
#2 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(32): call_user_func_array()
#3 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/Util.php(36): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#4 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(90): Illuminate\\Container\\Util::unwrapIfClosure()
#5 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(34): Illuminate\\Container\\BoundMethod::callBoundMethod()
#6 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/Container.php(590): Illuminate\\Container\\BoundMethod::call()
#7 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Console/Command.php(134): Illuminate\\Container\\Container->call()
#8 /var/www/snipeit/vendor/symfony/console/Command/Command.php(255): Illuminate\\Console\\Command->execute()
#9 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Console/Command.php(121): Symfony\\Component\\Console\\Command\\Command->run()
#10 /var/www/snipeit/vendor/symfony/console/Application.php(1001): Illuminate\\Console\\Command->run()
#11 /var/www/snipeit/vendor/symfony/console/Application.php(271): Symfony\\Component\\Console\\Application->doRunCommand()
#12 /var/www/snipeit/vendor/symfony/console/Application.php(147): Symfony\\Component\\Console\\Application->doRun()
#13 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Console/Application.php(93): Symfony\\Component\\Console\\Application->run()
#14 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(131): Illuminate\\Console\\Application->run()
#15 /var/www/snipeit/artisan(35): Illuminate\\Foundation\\Console\\Kernel->handle()
#16 {main}
"}
```

**Comment by snipe** [comment]
Can you do another pull from develop and see if that fixes your issue?

**Comment by denzfarid** [comment]
nope, same issue 
this i checkout to branch develop and i pull

<img width="873" alt="image" src="https://user-images.githubusercontent.com/1294403/99010664-04680e00-257d-11eb-8306-aed74151c06a.png">

<img width="344" alt="image" src="https://user-images.githubusercontent.com/1294403/99010705-16e24780-257d-11eb-914f-b7eb026b4ac8.png">


<img width="1602" alt="image" src="https://user-images.githubusercontent.com/1294403/99010721-23ff3680-257d-11eb-9157-6b365ce8c097.png">


```[2020-11-13 06:54:10] production.ERROR: Undefined offset: 2 {"exception":"[object] (ErrorException(code: 0): Undefined offset: 2 at /var/www/snipeit/app/Console/Commands/MoveUploadsToNewDisk.php:94)
[stacktrace]
#0 /var/www/snipeit/app/Console/Commands/MoveUploadsToNewDisk.php(94): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError()
#1 [internal function]: App\\Console\\Commands\\MoveUploadsToNewDisk->handle()
#2 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(32): call_user_func_array()
#3 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/Util.php(36): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#4 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(90): Illuminate\\Container\\Util::unwrapIfClosure()
#5 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(34): Illuminate\\Container\\BoundMethod::callBoundMethod()
#6 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Container/Container.php(590): Illuminate\\Container\\BoundMethod::call()
#7 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Console/Command.php(134): Illuminate\\Container\\Container->call()
#8 /var/www/snipeit/vendor/symfony/console/Command/Command.php(255): Illuminate\\Console\\Command->execute()
#9 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Console/Command.php(121): Symfony\\Component\\Console\\Command\\Command->run()
#10 /var/www/snipeit/vendor/symfony/console/Application.php(1001): Illuminate\\Console\\Command->run()
#11 /var/www/snipeit/vendor/symfony/console/Application.php(271): Symfony\\Component\\Console\\Application->doRunCommand()
#12 /var/www/snipeit/vendor/symfony/console/Application.php(147): Symfony\\Component\\Console\\Application->doRun()
#13 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Console/Application.php(93): Symfony\\Component\\Console\\Application->run()
#14 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(131): Illuminate\\Console\\Application->run()
#15 /var/www/snipeit/artisan(35): Illuminate\\Foundation\\Console\\Kernel->handle()
#16 {main}
"}````

**Comment by snipe** [comment]
Goddammit. Okay, I just pushed out a few more fixes for that script.

**Comment by denzfarid** [comment]
> Goddammit. Okay, I just pushed out a few more fixes for that script.

Fantastic !!, I really appreciate it, the command is perfect, and there are no errors

i am very very thank you
<img width="342" alt="image" src="https://user-images.githubusercontent.com/1294403/99046980-1b7f1e00-25c6-11eb-9a3e-73aaf1c0ae09.png">


this for image profile

<img width="201" alt="image" src="https://user-images.githubusercontent.com/1294403/99060717-02c93500-25d3-11eb-921d-cfa3f9f710a2.png">

**Comment by snipe** [comment]
Glad that worked for you. Please do keep us apprised of any other weirdness you find with the S3 stuff. It's still very much a work in progress.

**Comment by denzfarid** [comment]
> Glad that worked for you. Please do keep us apprised of any other weirdness you find with the S3 stuff. It's still very much a work in progress.

okay, I haven't tried this s3 more, quick question, 
what is the difference between public and private? what is the difference between public and private?

I just tried a fresh update (I deleted existing assets on S3, and I ran again),

In this latest update, I only found private uploads and no public uploads.

<img width="504" alt="image" src="https://user-images.githubusercontent.com/1294403/99082333-9ceea480-25f6-11eb-824b-ffb705f73c1c.png">

prior to this, public assets were uploaded

this is the bucket list, once uploaded
<img width="153" alt="image" src="https://user-images.githubusercontent.com/1294403/99082680-0bcbfd80-25f7-11eb-8e59-23f016503526.png">
